### DISCUSSION TITLE
Actix-web framework master

### DISCUSSION THREAD

**Discussion by QaisarAzeemi** [tokio-rs/tokio#3855] [General]
Hi,
I want to master Actix-web framework for Rust by building and developing different projects. I searched it on google but couldn't find any project to work and reproduce for learning purpose. I need your kind help in this regard.
my second question is that is actix web and tokio.rs and hyper are used for same purposes (Different paths to reach same destination) or they have different use?
Thanking you in anticipation :)
M Qaisar Azeemi

**Comment by Darksonn** [comment]
Check out the actix [getting started guide][1]. If that is not enough, then I have heard that [Zero To Production][2] is really good.

As for the three things you listed, they are not for the same purposes. Actix-web is a higher level framework for web servers. Hyper is a low-level library for web servers, but you would usually not use hyper directly — instead actix-web would use hyper (or something similar). 

Tokio is an asynchronous runtime, and you can think of it as the standard library for asynchronous Rust code.

[1]: https://actix.rs/docs/getting-started/
[2]: https://www.zero2prod.com/index.html

**Reply by QaisarAzeemi** [reply]
Thank you very much dear darksonn. I am working on getting started guide but yes, it is not enough. Is there any alternative to zero to production? I checked on google but couldn't found.

**Reply by Darksonn** [reply]
There are probably various blog posts on the web for specific topics, but I don't know about them. You could try to ask [here](https://discord.gg/NWpN5mmg3x).
